src/audio/google_audio_post_processing.c

```diff
@@ -0,0 +1,551 @@
+			cdata->cmd);
+		return -EINVAL;
+	}
+}
+
+static int gapp_set_comp_config(struct comp_dev *dev,
+				struct sof_ipc_ctrl_data *cdata)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	if (cdata->data->size != sizeof(cd->config)) {
+		comp_err(dev, "gapp_set_comp_config(): invalid data size %d",
+			 cdata->data->size);
+		return -EINVAL;
+	}
+
+	memcpy_s(cd->config, sizeof(cd->config), cdata->data->data,
+		 cdata->data->size);
+	comp_dbg(dev, "GAPP new settings c[0] %u c[1] %u", cd->config[0],
+		 cd->config[1]);
+
+	return 0;
+}
+
+static int gapp_ctrl_set_bin_data(struct comp_dev *dev,
+				  struct sof_ipc_ctrl_data *cdata)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	assert(cd);
+	comp_dbg(
+		dev,
+		"gapp_ctrl_set_bin_data(): type = %u, comp_id = %u, index = %u, blob_size = %d",
+		cdata->data->type, cdata->comp_id, cdata->msg_index,
+		cdata->num_elems);
+
+	if (dev->state < COMP_STATE_READY) {
+		comp_err(dev, "gapp_ctrl_set_bin_data(): driver in init!");
+		return -EBUSY;
+	}
+
+	switch (cdata->data->type) {
+	case SOF_GAPP_CONFIG:
+		return gapp_set_comp_config(dev, cdata);
+	case SOF_GAPP_MODEL:
+		return comp_data_blob_set_cmd(cd->tuning_handler, cdata);
+	case SOF_GAPP_DATA:
+		return 0;
+	default:
+		comp_err(dev,
+			 "gapp_ctrl_set_bin_data(): unknown binary data type");
+		return -EINVAL;
+	}
+}
+
+static int gapp_ctrl_set_data(struct comp_dev *dev,
+			      struct sof_ipc_ctrl_data *cdata)
+{
+	/* Check version from ABI header */
+	if (SOF_ABI_VERSION_INCOMPATIBLE(SOF_ABI_VERSION, cdata->data->abi)) {
+		comp_err(dev, "gapp_ctrl_set_data(): invalid version");
+		return -EINVAL;
+	}
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		return gapp_ctrl_set_bin_data(dev, cdata);
+	default:
+		comp_err(dev, "gapp_ctrl_set_data(): invalid cdata->cmd");
+		return -EINVAL;
+	}
+}
+
+static int gapp_get_comp_config(struct comp_data *cd,
+				struct sof_ipc_ctrl_data *cdata,
+				int max_data_size)
+{
+	if (sizeof(cd->config) > max_data_size)
+		return -EINVAL;
+
+	memcpy_s(cdata->data->data, max_data_size, cd->config,
+		 sizeof(cd->config));
+	cdata->data->abi = SOF_ABI_VERSION;
+	cdata->data->size = sizeof(cd->config);
+
+	return 0;
+}
+
+static int gapp_get_internal_config(struct comp_data *cd,
+				    struct sof_ipc_ctrl_data *cdata,
+				    int max_data_size)
+{
+	int data_size;
+	int blob_size = cdata->num_elems;
+
+	if (blob_size > max_data_size)
+		return -EINVAL;
+
+	data_size = GoogleAudioPostProcessingGetConfig(
+		cd->state, cdata->data->type, cdata->msg_index,
+		(uint8_t *)cdata->data->data, blob_size);
+
+	if (data_size < 0)
+		return -EINVAL;
+
+	cdata->data->abi = SOF_ABI_VERSION;
+	cdata->data->size = data_size;
+
+	return 0;
+}
+
+static int gapp_ctrl_get_bin_data(struct comp_dev *dev,
+				  struct sof_ipc_ctrl_data *cdata,
+				  int max_data_size)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	assert(cd);
+	comp_dbg(
+		dev,
```

Comment:
dev can be moved to the prev line

Reply:
Updated the printed info, and it's now moved to prev line.